src/vs/vscode.d.ts

```diff
@@ -4452,6 +4452,11 @@ declare module 'vscode' {
 		 * Args for the custom shell executable, this does not work on Windows (see #8429)
 		 */
 		shellArgs?: string[];
+		/**
+		 * A custom environment for the terminal, if this is not set the environment will be inherited
```

Comment:
We are indeed replacing the env. The `env` property on `IShellLaunchConfig` does eventually replace and not add to the VS Code process.

I can change the api to make it additive if you are ok with changing the use of `IShellLaunchConfig` as well